### PR TITLE
Delete related changes.

### PR DESCRIPTION
**Pull request by katmsft**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

File: sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_options.hpp [hunk 1]
```diff
@@ -661,13 +652,10 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
...
-     * @brief When deleting a directory, the number of paths that are deleted with each invocation
-     *        is limited. If the number of paths to be deleted exceeds this limit, a continuation
-     *        token is returned in this response header.  When a continuation token is returned in
-     *        the response, it must be specified in a subsequent invocation of the delete operation
-     *        to continue deleting the directory.
+     * @brief If "true", all paths beneath the directory will be deleted. If "false" and the
+     * directory is non-empty, an error occurs.
      */
-    Azure::Core::Nullable<std::string> ContinuationToken;
+    bool Recursive = false;
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_options.hpp:658]: I demand to revisit the default value

**Review by Jinming-Hu — CHANGES_REQUESTED** [review]

File: sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_options.hpp (hunk 1 above)
Comment (katmsft) [reply sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_options.hpp]: Revisited with Jeffrey, will use the new design.

File: sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_directory_client.hpp [hunk 2]
```diff
@@ -146,30 +146,56 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
         = RenameDataLakeSubdirectoryOptions()) const;
 
     /**
-     * @brief Deletes the directory.
-     * @param recursive If "true", all paths beneath the directory will be deleted. If "false" and
-     *                  the directory is non-empty, an error occurs.
+     * @brief Deletes the empty directory. Errors if directory is not empty.
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_directory_client.hpp:149]:
```suggestion
     * @brief Deletes the empty directory. Exception will be thrown if directory is not empty.
```

File: sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_directory_client.hpp [hunk 3]
```diff
@@ -146,30 +146,56 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
...
-        bool recursive,
-        const DeleteDataLakeDirectoryOptions& options = DeleteDataLakeDirectoryOptions()) const;
+    Azure::Core::Response<Models::DeleteDataLakeDirectoryResult> DeleteIfEmptyIfExists(
+        const DeleteDataLakeDirectoryOptions& options = DeleteDataLakeDirectoryOptions()) const
+    {
+      return this->DeleteIfExists(false, options);
+    }
+
+    /**
+     * @brief Deletes the directory and all its subfolders and files.
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_directory_client.hpp:175]:
```suggestion
     * @brief Deletes the directory and all its subdirectories and files.
```

File: sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_directory_client.hpp [hunk 4]
```diff
@@ -193,6 +219,17 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
...
+        bool recursive,
+        const DeleteDataLakeDirectoryOptions& options = DeleteDataLakeDirectoryOptions()) const;
+
+    Azure::Core::Response<Models::DeleteDataLakeDirectoryResult> DeleteIfExists(
+        bool recursive,
+        const DeleteDataLakeDirectoryOptions& options = DeleteDataLakeDirectoryOptions()) const;
+
+    // Hide path functions
+    using DataLakePathClient::Delete;
+    using DataLakePathClient::DeleteIfExists;
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_directory_client.hpp:232]: Do we still need these two lines? given you've declared them above

**Review by Jinming-Hu — APPROVED** [review]

**Pull request merged**